src/hooks/useAsyncStorage.ts

```diff
@@ -2,7 +2,7 @@ import AsyncStorage from '@react-native-async-storage/async-storage';
 import { useQuery } from 'react-query';
 
 export function useAsyncStorage(key: string) {
-  const itemResult = useQuery<string | null>('async-storage-get', () =>
+  const itemResult = useQuery<string | null>(['async-storage-get', key], () =>
     AsyncStorage.getItem(key),
   );
 
```

Comment:
This was a bug where `useQuery` global cache key was being set to `async-storage-get` for all Async Storage key/value pairs.  Adding the `key` to make it unique fixes it.  

I spent some time writing tests for this but eventually bailed on that because I'm not sure using `useQuery` here is a good solution anyway.

I would be in favor of replacing it with a light wrapper that abstracts away serialization/deserialization, maybe something [like this](https://github.com/infinitered/ignite/blob/master/boilerplate/app/utils/storage/storage.ts).  That way we could easily replace Async Storage with something much faster in the future if we want... [like this](https://github.com/mrousavy/react-native-mmkv).